prospero-common/src/main/java/org/wildfly/prospero/api/ProvisioningDefinition.java

```diff
@@ -164,29 +203,48 @@ public ProvisioningConfig toProvisioningConfig() throws ProvisioningException {
         }
     }
 
+    /**
+     * Resolves channel coordinates into Channel instances.
+     *
+     * @param versionResolverFactory a VersionResolverFactory instance to perform the channel resolution
+     * @return Channel instances
+     */
+    public List<Channel> resolveChannels(VersionResolverFactory versionResolverFactory) {
+        try {
+            List<Channel> channels = new ArrayList<>(this.channels);
+
+            if (!channelCoordinates.isEmpty()) {
+                channels.addAll(versionResolverFactory.resolveChannels(channelCoordinates, defaultChannelResolutionRepositories()));
```

Comment:
shouldn't channelCoordinates allow path as well as URL and GAV?

Reply:
If they were Paths, they were already converted to URLs at this point. The conversion is done in the `ProvisioningDefinition.Builder#setChannelCoordinates()`.

Reply:
ah right, missed that 👍